src-tauri/src/farmer.rs

```diff
@@ -59,43 +57,49 @@ pub(crate) async fn farming(path: String, reward_address: String, plot_size: u64
             dsn_sync: false,
         };
 
-        if let Err(err) = farm(path.clone().into(), farming_args.clone()).await {
-            error!("farm function failed to start, with error: {err}");
+        let farming_handle;
+        match farm(path.clone().into(), farming_args.clone()).await {
+            Err(error) => {
+                return Err(format!(
+                    "farm function failed to start, with error: {error}"
+                ))
+            }
+            Ok(handle) => farming_handle = handle,
         }
```

Comment:
```suggestion
        let farming_handle = farm(path.clone().into(), farming_args.clone())
            .await
            .map_err(|error| format!(
                "farm function failed to start, with error: {error}"
            ))?;
```

Reply:
thanks! will commit after Serge replies, maybe I will change it to `Result<(), String>` as you suggested,